src/library/tactic/user_attribute.cpp

```diff
@@ -117,6 +160,16 @@ vm_obj attribute_fingerprint(vm_obj const & vm_n, vm_obj const & vm_s) {
     return mk_tactic_success(mk_vm_nat(h), s);
 }
 
+vm_obj caching_user_attribute_get_cache(vm_obj const & vm_attr, vm_obj const & vm_s) {
+    auto const & s = to_tactic_state(vm_s);
+    auto const & n = to_name(cfield(vm_attr, 0));
+    auto const & cache_handler = cfield(vm_attr, 2);
```

Comment:
Finding out the field index was trial and error, I'm not quite clear yet at what point the `Cache` field / the dependent pair vanishes.